### ISSUE TITLE
Per-pixel/per-element-loss of a tensor/matrix

### ISSUE THREAD

**Issue by Mut1nyJD** [Microsoft/CNTK#3315]
What would be the best way to implement a per-pixel / per-element loss in CNTK at the moment? 
As far as I could gather from the python documentation nearly all the loss function expect a vector-type shape? So would you have to reshape the tensor to a vector and then evaluate the loss and then shape the gradients back to the original tensor format?

Or is there another way of doing it?

**Comment by delzac** [comment]
Hi,

Actually there's a [PR](https://github.com/Microsoft/CNTK/pull/3284/files) that may be what you are looking for.

fmeasure is used for semantic segmentation, i.e. whether this particular pixel is a car/road/building/etc.

Typically, if you don't have imbalance class, you should be able to use binary_cross_entropy or cross_entropy_with_softmax (you can't use cntk's implementation as it expects one vector per sample and in semantic segmentation you have multiple vectors per sample, as many vectors as pixel. However, implementing one on your own cross entropy for semantic segmentation should be trivial)

If you only have binary class, you can use cntk's implementation as is.

**Comment by YanYas** [comment]
Hi @delzac, did f-measure work for the semantic segmentation task? I noticed its something that has been mentioned as work-in-progress on a few issues. Did this loss function bring you any success and if so, with which network (UNet, FCN etc)?

**Comment by delzac** [comment]
Yup, it worked for me. It used it with UNET and [FC-DenseNet](https://arxiv.org/abs/1611.09326). In #3396 , i added a suggestion in the docs on an easier form of the loss function to optimise. Basically you log() it.

**Comment by YanYas** [comment]
Great to hear. I'll have to feedback here in a couple days time, but I'll endeavour to do so
